src/components/tooltip/tooltip.js

```diff
@@ -187,6 +187,9 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
     }
 
     function setVisible (value) {
+      if ((setVisible.queued && setVisible.visible === !!value) || (!setVisible.queued && scope.visible === !!value)) {
```

Comment:
@ThomasBurleson by the way, this perf issue made me disable tooltips in my company's project for the time being. I would be glad if we could push some kind of perf fix here as soon as possible.